platformio.ini

```diff
@@ -22,6 +22,11 @@ board = lolin32
 framework = arduino
 lib_ldf_mode = deep+
 monitor_speed = 115200
+upload_protocol = espota
+upload_port = 192.168.20.1
+upload_flags =
+    --host_ip=192.168.20.2
```

Comment:
This is not guaranteed to be the host IP, e.g. if you connect with 2 devices, the 2nd device will get `192.168.20.3` as the IP address. Are you sure this parameter is required?

Reply:
You are right. I had issues due to working on Virtual Machine, but it should not be mandatory. I change it to comment as it might be helpful for someone.